channeldb/payments.go

```diff
@@ -957,6 +927,78 @@ func (d *DB) DeletePayments(failedOnly, failedHtlcsOnly bool) error {
 	}, func() {})
 }
 
+// deletePaymentAttempt removes an HTLC attempt from disk and returns the
+// payment in the state before it was deleted.
+func (d *DB) deletePaymentAttempt(hash lntypes.Hash, attemptID uint64) (*MPPayment, error) {
+	aid := make([]byte, 8)
+	binary.BigEndian.PutUint64(aid, attemptID)
+
+	var payment *MPPayment
+	err := kvdb.Update(d, func(tx kvdb.RwTx) error {
+		payment = nil
+		payments := tx.ReadWriteBucket(paymentsRootBucket)
+		if payments == nil {
+			return nil
```

Comment:
I think we need to return an error here and below if buckets cannot be found.

Reply:
Done. And I updated the function header to reflect this.